pkg/backup/backup.go

```diff
@@ -408,6 +409,24 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
 	}
 	backupRequest.Status.Progress = &velerov1api.BackupProgress{TotalItems: len(items)}
 
+	// Build PVC-to-Pod cache for improved volume policy lookup performance.
+	// This avoids O(N*M) complexity when there are many PVCs and pods.
+	// See issue #9179 for details.
+	pvcPodCache := podvolumeutil.NewPVCPodCache()
```

Comment:
Since this `pvcPodCache` is used by `volumehelper` only, why don't we move this cache build into volumehelper? Benefits:
1. All the logics are in `volumehelper`, no need to change the backupper
2. `volumehelper` may be per backup, or it is also reasonable per backups which is started in the same period of time, having the cache in `volumehelper` ensures the cache always have the same lifecycle with the `volumehelper`